contribution_to_project.rst

```diff
@@ -0,0 +1,95 @@
+================================================
+
+A contribution to an Open Source project is any good given by an individual or a company to the project including increasing its functionality, improvements of the codebase, code developer experience, user experience, increasing the project's popularity and community, providing infrastructure, and any other kinds of active social or material support.
+
+Most the types of contribution to an Open Source project do not require programming skills.
+
+Classification of contributions
+===============================
+
+Contributions can be one of the following types:
+
+  - :ref:`Contribution to a codebase<Contribution to codebase>`.
+  - :ref:`Contribution to code developer experience<Contribution to dev exp>`.
+  - :ref:`Contribution to user experience<Contribution to user exp>`.
+  - :ref:`Contribution to a community<Contribution to community>`.
+  - :ref:`Other kinds of contributions<Other kinds of contributions>`.
+
+.. _Contribution to codebase:
+
+Contribution to a codebase
+--------------------------
+
+Contribution to software itself (its codebase) insludes:
+
+  - Proposing new features.
+  - Reporting bugs.
+  - Participating in design and implementation of new components.
+  - Bug fixes.
+  - Code reviewing.
+  - Code refactoring.
+  - Improving stability of an existing codebase through improving a testing codebase and infrastructure.
+  - Maintenance (for example, backporting and releasing).
+
+Examples of codebase contributions which do not require programming skills in scope of the Ansible project can be reviewing proposals, testing pull requests locally, adding integration tests, bug reports, feature requests, providing user experience to programmers in issues and pull requests.
+
+.. _Contribution to dev exp:
+
+Contribution to code developer experience
+-----------------------------------------
+
+Contribution to code developer experience includes sharing experience through improving code development documentation, writing development guidelines, mentoring and training code contributors.
+
+.. _Contribution to user exp:
+
+Contribution to user experience
+-------------------------------
+
+Contribution to user experience includes sharing experience about how to use software more efficiently through improving user documentation, writing manuals and books, making videos, courses, and so on.
+
+Examples:
+
+  - Clarification and updating user documentation.
+  - Making a presentation for a local meetup covering not well-known but useful usage aspects.
+
+.. _Contribution to community:
+
+Contribution to a community
+-----------------------------
+
+In a narrow sense, contribution to a community includes:
+
+  - Promoting the project among people.
+  - Thinking of and proposing ideas about how community processes can be improved.
+  
+Examples:
+
+  - Organizing a meetup.
+  - Writing a blogpost about benefits of the project comparing with other similar projects.
```

Comment:
There is a reason other similar projects exist, if a blog post is all talking good about a project I do not consider it helpful really.
```suggestion
  - Writing a blog post about benefits (and possible drawbacks) of the project comparing other similar projects.
```

Reply:
I'm not native English speaker, but it seems that comparing to is not right (https://dictionary.cambridge.org/dictionary/english/comparing).

Reply:
sounds sensible, changed. though the balance is important if we want to see community growth, not a decline:D

Reply:
> I'm not a native English speaker, but it seems that "comparing to" is not right (https://dictionary.cambridge.org/dictionary/english/comparing).

Changed, thanks!